lib/ide/src/view/layout.rs

```diff
@@ -136,43 +104,39 @@ impl ViewLayoutData {
         self.text_editor.set_padding(padding);
         self.text_editor.set_size(size);
         self.text_editor.set_position(position);
-        self.text_editor.update();
     }
 }
 
 impl ViewLayout {
     /// Creates a new ViewLayout with a single TextEditor.
-    pub fn default(world:&World) -> Self {
-        let text_editor  = TextEditor::new(&world);
-        let key_listener = None;
+    pub fn new
+    ( logger               : &Logger
+    , kb_actions           : &mut KeyboardActions
+    , world                : &World
+    , controller           : controller::text::Handle
+    ) -> Self {
+        let logger       = logger.sub("ViewLayout");
+        let text_editor  = TextEditor::new(&logger,&world,controller,kb_actions);
         let layout_mode  = default();
         let size         = zero();
-        let data         = ViewLayoutData {text_editor,key_listener,layout_mode,size};
+        let data         = ViewLayoutData {text_editor,layout_mode,size,logger};
         let rc           = Rc::new(RefCell::new(data));
-        Self {rc}.init(world)
+        Self {rc}.init(world,kb_actions)
     }
 
-    fn init_keyboard(self) -> Self {
-        let view_layout = self.clone();
-        let closure     = move |event:KeyboardEvent| {
-            const F_KEY : u32 = 70;
-            if event.ctrl_key() && event.key_code() == F_KEY {
-                view_layout.switch_layout_mode();
-                event.prevent_default();
-            }
-        };
-        let closure : Box<dyn FnMut(KeyboardEvent)> = Box::new(closure);
-        let callback                                = Closure::wrap(closure);
-        let body                                    = document().unwrap().body().unwrap();
-        let key_listener = KeyboardListener::new(&body, "keydown".into(), callback);
-        self.rc.borrow_mut().key_listener = Some(key_listener);
+    fn init_keyboard(self, keyboard_actions:&mut KeyboardActions) -> Self {
+        let save_keys:KeyMask = [Key::Control, Key::Character("f".to_string())].iter().collect();
```

Comment:
Is the name correct? `save_keys`
Should be rather about switching layout mode

Reply:
Also, I'd consider having some helper to create KeyMask for "Control+Key", it currently looks like a handful and will continue reappear.

Reply:
I introduced:

```rust
    /// Creates Key::Control + Key::Character.
    pub fn new_control_character(character:char) -> Self {
        Self::from_vec(vec![Key::Control, Key::Character(character.to_string())])
    }

    /// Creates KeyMask from Vec<Key>.
    pub fn from_vec(keys:Vec<Key>) -> Self {
        keys.iter().collect()
    }
```